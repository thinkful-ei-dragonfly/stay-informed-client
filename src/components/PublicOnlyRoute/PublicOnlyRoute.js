import React from 'react'
import { Route, Redirect } from 'react-router-dom'
import UserContext from '../../contexts/UserContext'

export default function PublicOnlyRoute({ component, ...props }) {
  const Component = component
  return (
    <Route
      {...props}
      render={componentProps => (
        <UserContext.Consumer>
          {userContext =>
            !!userContext.user.id
              ? <Redirect to={'/'} />
              : <Component {...componentProps} />
          }
        </UserContext.Consumer>
      )}
    />
  )
}
