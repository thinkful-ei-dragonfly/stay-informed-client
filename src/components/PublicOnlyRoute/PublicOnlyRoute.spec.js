import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import PublicOnlyRoute from './PublicOnlyRoute'

describe(`PublicOnlyRoute component`, () => {
  it('renders a form.PublicOnlyRoute by default', () => {
    const wrapper = shallow(<PublicOnlyRoute />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

})
