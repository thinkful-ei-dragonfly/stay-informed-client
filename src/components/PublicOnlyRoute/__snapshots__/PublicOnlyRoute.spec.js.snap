// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicOnlyRoute component renders a form.PublicOnlyRoute by default 1`] = `
<Route
  render={[Function]}
/>
`;
